src/sentry/core/endpoints/organization_details.py

```diff
@@ -558,6 +588,64 @@ def save(self, **kwargs):
         if trusted_relay_info is not None:
             self.save_trusted_relays(trusted_relay_info, changed_data, org)
 
+        if "hasGranularReplayPermissions" in data:
+            option_key = "sentry:granular-replay-permissions"
+            new_value = data["hasGranularReplayPermissions"]
+            option_inst, created = OrganizationOption.objects.update_or_create(
+                organization=org, key=option_key, defaults={"value": new_value}
+            )
```

Comment:
You could use `org.update_option(option_key, new_value)` to insert/update the option value.